solver/jobs.go

```diff
@@ -901,9 +903,12 @@ func (s *sharedOp) LoadCache(ctx context.Context, rec *CacheRecord) (Result, err
 	// no cache hit. start evaluating the node
 	span, ctx := tracing.StartSpan(ctx, "load cache: "+s.st.vtx.Name(), trace.WithAttributes(attribute.String("vertex", s.st.vtx.Digest().String())))
 	notifyCompleted := notifyStarted(ctx, &s.st.clientVertex, true)
-	res, err := s.Cache().Load(withAncestorCacheOpts(ctx, s.st), rec)
+	res, err := s.Cache().Load(withAncestorCacheOpts(ctx, s), rec)
 	tracing.FinishWithError(span, err)
 	notifyCompleted(err, true)
+	if err == nil {
+		s.loadCacheOpts = res.CacheOpts()
```

Comment:
Can't we just use interface detection on `res.Sys()` in here?

Reply:
Or maybe `Load()` should return it separately? Something seems off about having `CacheOpts()` method in the `Result` interface.